src/editor.ts

```diff
@@ -73,55 +111,41 @@ export class Editor {
 	}
 
 	// Kill to end of line
-	async kill(): Promise<boolean> {
+	killForward(): void {
 		// Ignore whatever we have selected before
-		await vscode.commands.executeCommand("emacs.exitMarkMode")
-
-		let startPos = this.getCurrentPos(),
-			isOnLastLine = Editor.isOnLastLine()
-
-		// Move down an entire line (not just the wrapped part), and to the beginning.
-		await vscode.commands.executeCommand("cursorMove", { to: "down", by: "line", select: false })
-		if (!isOnLastLine) {
-			await vscode.commands.executeCommand("cursorMove", { to: "wrappedLineStart" })
-		}
-
-		let endPos = this.getCurrentPos(),
-			range = new vscode.Range(startPos, endPos),
-			txt = vscode.window.activeTextEditor.document.getText(range)
-
-		// If there is something other than whitespace in the selection, we do not cut the EOL too
-		if (!isOnLastLine && !txt.match(/^\s*$/)) {
-			await vscode.commands.executeCommand("cursorMove", {to: "left", by: "character"})
-			endPos = this.getCurrentPos()
-		}
-
-		// Select it now, cut the selection, remember the position in case of multiple cuts from same spot
-		this.setSelection(startPos, endPos)
-		let promise = this.cut(this.lastKill != null && startPos.isEqual(this.lastKill))
-
-		promise.then(() => {
-			this.justDidKill = true
-			this.lastKill = startPos
+		vscode.commands.executeCommand("emacs.exitMarkMode")
+		// move cursor to the end of the line and select the text
+		vscode.commands.executeCommand("cursorMove", {
+			to:"wrappedLineEnd",
+			select:true
 		})
+		// editor.action.clipboardCutAction is too slow!
+		vscode.commands.executeCommand("editor.action.clipboardCopyAction");
+		vscode.commands.executeCommand("deleteAllRight");
+	}
 
-		return promise
+	// Kill to beginning of line
+	killBackward(): void {
+		// Ignore whatever we have selected before
+		vscode.commands.executeCommand("emacs.exitMarkMode")
+		// move cursor to the beginning of the line and select the text
+		vscode.commands.executeCommand("cursorMove", {
+			to:"wrappedLineStart",
+			select:true
+		})
+		// editor.action.clipboardCutAction is too slow!
+		vscode.commands.executeCommand("editor.action.clipboardCopyAction");
+		vscode.commands.executeCommand("deleteAllLeft");
 	}
 
 	copy(): void {
-		clip.writeSync(this.getSelectionText())
+		vscode.commands.executeCommand("editor.action.clipboardCopyAction")
 		vscode.commands.executeCommand("emacs.exitMarkMode")
 	}
 
-	cut(appendClipboard?: boolean): Thenable<boolean> {
-		if (appendClipboard) {
-			clip.writeSync(clip.readSync() + this.getSelectionText());
-		} else {
-			clip.writeSync(this.getSelectionText());
-		}
-		let t = Editor.delete(this.getSelectionRange());
-		vscode.commands.executeCommand("emacs.exitMarkMode");
-		return t
+	cut(): void {
+		vscode.commands.executeCommand("editor.action.clipboardCutAction")
+		Editor.inMarkMode = false
```

Comment:
Strangely if you permorf a "cancelSelection" after the "clipboardCutAction" the selected region will be cutted, but the hole line will disappear, probably it's a vscode bug. Tried to use async-await with no success.